src/session.rs

```diff
@@ -0,0 +1,83 @@
+extern crate dirs;
+
+use crate::auth::Auth;
+use crate::request_items::Parameter;
+use crate::utils::ensure_session_dir_exists;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::io::prelude::*;
+
+const VERSION: &'static str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Session {
+    pub identifier: String,
+    pub host: String,
+    pub headers: Vec<Parameter>,
+    pub auth: Option<Auth>,
+    ht_version: String,
+}
+
+impl Session {
+    pub fn new(identifier: String, host: &str, headers: Vec<Parameter>, auth: Option<Auth>) -> Session {
```

Comment:
If we're going to pay the cost of converting `host` to a `String` unconditionally, I believe we should have the calling code do that explicitly (which'd mean we'd use `host: String` here) instead of hiding it from the user.

Reply:
Ok, got it.